tests/plugins/widget/copycutintegration.js

```diff
@@ -0,0 +1,70 @@
+/* bender-tags: widget, clipbaord */
+/* bender-ui: collapsed */
+/* bender-ckeditor-plugins: widget, undo, wysiwygarea, toolbar, clipboard */
+
+( function() {
+	'use strict';
+
+	CKEDITOR.plugins.add( 'customwidget', {
+		requires: 'widget',
+		allowedContent: 'div',
+
+		init: function( editor ) {
+			editor.widgets.add( 'customwidget', {
+				button: 'Add widget',
+				template: '<div>Widget</div>'
+			} );
+		}
+	} );
+
+	bender.editor = {
+		name: 'editor',
+		startupData: '<p>foo bar</p><div data-widget="customwidget">Widget</div>',
+		config: {
+			extraAllowedContent: 'span',
+			extraPlugins: 'customwidget',
+			allowedContent: true
+		}
+	};
+
+	bender.test( {
+		'test widget not throw error during copy': function() {
+			var editor = this.editor,
+				editable = this.editor.editable(),
+				range = editor.createRange();
+
+			range.setStart( editable.findOne( 'p' ).getFirst(), 4 );
+			range.setEnd( editable.findOne( 'div' ), 0 );
+			range.select();
+
+			editable.fire( 'copy', new CKEDITOR.dom.event( {} ) );
+			assert.pass();
+		},
+
+		'test cutting partially selected widget remain collapsed selection in editor': function() {
```

Comment:
I would be for adding one more test, similar to this one but testing more complex widgets (e.g. with nested editable in which selection starts/ends).

Reply:
Shouldn't this test also check what was copied/cut (or maybe you need separate test for that)?

Reply:
I tried to achieve that, however, the way how the copybin is working doesn't allow on that.
Single tests work fine, but when there are multiple editors, running one test after another the problematic situation starts to appear. The copybin is destroyed asynchronously and there is no easy way to track it. :(
I feel that still, those tests require improvements, however, I don't see any easy nice way to properly attach to it.